spec/lib/zendesk_apps_tools/deploy_spec.rb

```diff
@@ -6,46 +6,56 @@
   let(:subject_class) { Class.new { include ZendeskAppsTools::Deploy } }
 
   describe '#find_app_id' do
-    let(:api_response_with_app_ids) do
-     {
-       'apps'=> [
-          { 'name'=> 'notification_app', 'id'=> 22 },
-          { 'name'=> 'time_tracking_app', 'id' => 99 }
+    let (:api_response_with_installations) {
+      {
+        installations: [
+          { id: 1, app_id: 22, settings: { name: 'notification_app' }},
+          { id: 2, app_id: 99, settings: { name: 'time_tracking_app' }}
         ]
       }.to_json
-    end
+    }
 
-    define_method(:mocked_instance_methods_and_api) do |app_name|
+    def mocked_instance_methods_and_api(app_name, api_response)
       subject = subject_class.new
       allow(subject).to receive(:say_status)
       allow(subject).to receive(:get_value_from_stdin) { app_name }
-      allow(subject).to receive_message_chain(:cached_connection, :get, :body) { api_response_with_app_ids }
-
+      allow(subject).to receive_message_chain(:cached_connection, :get, :body) { api_response }
       subject
     end
 
+    context 'is unable to get a valid api response' do
+      it 'errors and exits system' do
+        invalid_api_response = {}
+        subject = mocked_instance_methods_and_api('notification_app', invalid_api_response)
+
+        expect(subject).to receive(:say_error).with(
+          "Unable to retrieve installations. Please check your internet connection."
+        )
+        expect { subject.find_app_id }.to raise_error(SystemExit)
+      end
+    end
+
     context 'user inputs an app name that is NOT in api response' do
       it 'errors and exits system' do
-        subject = mocked_instance_methods_and_api('random_app_name')
+        subject = mocked_instance_methods_and_api('random_app_name', api_response_with_installations)
 
         expect(subject).to receive(:say_error).with(
-          "App not found. " \
-          "Please verify that your credentials, subdomain, and app name are correct."
+          "App not found. Please verify that your credentials, subdomain, and app name are correct."
         )
         expect { subject.find_app_id }.to raise_error(SystemExit)
       end
     end
 
     context 'user inputs an app name that is in api response' do
       it 'returns app id 22 for notification_app' do
-        subject = mocked_instance_methods_and_api('notification_app')
+        subject = mocked_instance_methods_and_api('notification_app', api_response_with_installations)
         allow(subject).to receive_message_chain(:cache, :save)
 
         expect(subject.find_app_id).to eq(22)
       end
 
       it 'returns app id 99 for notification_app' do
```

Comment:
There's some confusion here between the 2 `it` blocks and the descriptive string that follows. "returns app id 99 for **notification_app**" and `mocked_instance_methods_and_api('**time_tracking_app**')`. `#Fixme`?

Reply:
this was my mistake sorry. I wrote the previous pr and i accidentally left this in it.

Reply:
Good catch, thanks